src/modules/airdrop/services/airdrop-service.ts

```diff
@@ -0,0 +1,196 @@
+      },
+      earlyUserRewards: {
+        eligible: !!walletRewards && new BigNumber(walletRewards.earlyUserRewards).gt(0),
+        amount: walletRewards?.earlyUserRewards || "0",
+      },
+      liquidityProviderRewards: {
+        eligible: !!walletRewards && new BigNumber(walletRewards.liquidityProviderRewards).gt(0),
+        amount: walletRewards?.liquidityProviderRewards || "0",
+      },
+      communityRewards: {
+        eligible: communityRewards ? new BigNumber(communityRewards).gt(0) : false,
+        amount: communityRewards || "0",
+      },
+    };
+  }
+
+  async processUploadedRewardsFiles({
+    walletRewardsFile,
+    communityRewardsFile,
+  }: {
+    walletRewardsFile?: Express.Multer.File;
+    communityRewardsFile?: Express.Multer.File;
+  }) {
+    if (communityRewardsFile) {
+      await this.processCommunityRewardsFile(communityRewardsFile);
+    }
+
+    if (walletRewardsFile) {
+      await this.processWalletRewardsFile(walletRewardsFile);
+    }
+
+    const [communityRewardsCount, walletRewardsCount] = await Promise.all([
+      this.communityRewardsRepository.count(),
+      this.walletRewardsRepository.count(),
+    ]);
+
+    return {
+      communityRewardsCount,
+      walletRewardsCount,
+    };
+  }
+
+  private async processWalletRewardsFile(walletRewardsFile: Express.Multer.File) {
+    try {
+      await this.walletRewardsRepository.update({ id: Not(IsNull()) }, { processed: false });
+      const walletRewardsContent = await readFile(walletRewardsFile.path, { encoding: "utf8" });
+      const walletRewards = JSON.parse(walletRewardsContent);
+
+      for (const walletAddress of Object.keys(walletRewards)) {
+        await this.insertWalletRewards({
+          walletAddress: ethers.utils.getAddress(walletAddress),
+          earlyUserRewards: walletRewards[walletAddress]["bridgoor"] || 0,
+          liquidityProviderRewards: walletRewards[walletAddress]["lp"] || 0,
+          welcomeTravellerRewards: walletRewards[walletAddress]["bridge-traveler"] || 0,
+        });
+      }
+    } catch (error) {
+      this.logger.error(error);
+      await this.walletRewardsRepository.update({ id: Not(IsNull()) }, { processed: true });
+      throw new ProcessWalletRewardsFileException();
+    }
+  }
+
+  private async processCommunityRewardsFile(communityRewardsFile: Express.Multer.File) {
+    try {
+      await this.communityRewardsRepository.update({ id: Not(IsNull()) }, { processed: false });
+
+      const communityRewardsContent = await readFile(communityRewardsFile.path, { encoding: "utf8" });
+      const communityRewards = JSON.parse(communityRewardsContent);
+
+      for (const communityReward of communityRewards) {
+        if (communityReward["ID"] && communityReward["Total Tokens"]) {
+          await this.insertCommunityRewards(communityReward["ID"], communityReward["Total Tokens"]);
+        }
+      }
+      await this.communityRewardsRepository.delete({ processed: false });
```

Comment:
why do we need to delete here?

Reply:
the JSON files contains the rewards, but there is no way to know in advance if a particular address/discord id was removed comparing to the previous rewards file. So before inserting/updating the rewards I set the existing rows to be `processed = false` and on insert/update they become `processed = true`. If any rows are left with `processed = false`, it means that they are not eligible for rewards anymore and they should be deleted